### PR TITLE
Add i18n support

### LINKED ISSUE #859: Add i18n
### Discussed in https://github.com/inovex/scrumlr.io/discussions/324

As an international customer I want an interface in my native language and I'm willing to translate all texts in Scrumlr.

AC:
- Add i18n support and extract all current text snippets into a default configuration
- Add configuration of interface language
- Document how new languages can be implemented

### PR DESCRIPTION
**Pull request by bitionaire**
Closes #859

### CODE REVIEW

File: public/locales/de/translation.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  },
+  "AccessPolicySelection": {
+    "title": "Zugriffsberechtigung",
+    "passphrase": "Passwort",
+    "public": "Zugriff für alle über die URL möglich.",
+    "byPassphrase": "Zugriff über die URL wird per Passwort geschützt.",
+    "manualVerification": "Moderatoren können Teilnehmer akzeptieren und ablehnen, ehe sie einer Sitzung beitreten.",
+    "generatePassphrase": "Erstelle zufälliges Passwort",
+    "copyPassphraseToClipboard": "Kopiere Passwort in Zwischenablage",
+    "passphraseValidationError": "Ein Passwort muss gesetzt sein"
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:34]:
```suggestion
    "passphraseValidationError": "Ein Passwort muss gesetzt sein!"
```

File: public/locales/de/translation.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  },
+  "BoardHeader": {
+    "showParticipants": "Zeige Teilnehmer"
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:37]:
```suggestion
    "showParticipants": "Zeige die Teilnehmer"
```

File: public/locales/de/translation.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  },
+  "ParticipantsList": {
+    "title": "Teilnehmer",
+    "search": "Suche",
+    "headerLegendName": "Name",
+    "headerLegendAdmin": "Admin"
+  },
+  "DeleteBoardOption": {
+    "warning": "Bist du dir sicher dass du die Sitzung löschen möchtest? Die Aktion ist nicht umkehrbar.",
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:46]:
```suggestion
    "warning": "Bist du dir sicher, dass du die Sitzung löschen möchtest? Die Aktion ist nicht umkehrbar.",
```

File: public/locales/de/translation.json [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  },
+  "JoinRequest": {
+    "title": "Jemand möchte der Sitzung beitreten",
+    "accept": "Akzeptieren",
+    "reject": "Ablehnen",
+    "acceptAll": "Alle akzeptieren",
+    "rejectAll": "Alle ablehnen"
+  },
+  "LoginProviders": {
+    "signInWithGoogle": "Sign in mit Google",
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:82]:
```suggestion
    "signInWithGoogle": "Einloggen mit Google",
```

File: public/locales/de/translation.json [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    "signInWithGitHub": "Sign in mit Github",
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:83]:
```suggestion
    "signInWithGitHub": "Einloggen mit Github",
```

File: public/locales/de/translation.json [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    "signInWithMicrosoft": "Sign in mit Microsoft"
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:84]:
```suggestion
    "signInWithMicrosoft": "Einloggen mit Microsoft"
```

File: public/locales/de/translation.json [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    "unmarkAsDone": "Als nicht fertig markieren",
+    "addImage": "Füge Bild hinzu",
+    "addSticker": "Füge Sticker hinzu",
+    "settings": "Einstellungen",
+    "editColumns": "Bearbeite Spalten",
+    "stopEditingColumns": "Beende Spaltenbearbeitung",
+    "startVoting": "Starte Abstimmung",
+    "stopVoting": "Beende Abstimmung",
+    "startFocusMode": "Starte Fokus-Modus",
+    "stopFocusMode": "Beende Fokus-Modus"
```
Comment (Dominik-Weinzierl) [inline public/locales/de/translation.json:121]: I am not sure if we should add articles in German or leave it like this to be similar to the English version

File: src/components/BoardHeader/HeaderMenu/BoardOptions/DeleteBoardOption.tsx [hunk 8]
```diff
@@ -23,15 +26,13 @@ export const DeleteBoardOption = (props: DeleteProps) => {
     <BoardOption data-testid="delete">
       <BoardOptionButton label="Delete board" icon={DeleteIcon} isExpandable onClick={props.onClick} />
```
Comment (Dominik-Weinzierl) [inline src/components/BoardHeader/HeaderMenu/BoardOptions/DeleteBoardOption.tsx:27]:
```suggestion
      <BoardOptionButton label={t("DeleteBoardOption.button")} icon={DeleteIcon} isExpandable onClick={props.onClick} />
```

**Review by Dominik-Weinzierl — COMMENTED** [review]
#### Some minor issues with German text:
Screenshot 2021-11-08 at 09 16 54

Screenshot 2021-11-08 at 09 16 17

#### And i think something is wrong with the padding here (but only in German):
Screenshot 2021-11-08 at 09 22 08

File: public/locales/de/translation.json (hunk 7 above)
Comment (bitionaire) [reply public/locales/de/translation.json:121]: It'll be too long for our menu bars 😬

Comment (Dominik-Weinzierl) [reply public/locales/de/translation.json:121]: Yes you are right. So we can leave it like this 👍

**Pull request merged**